src/validation/validateRequiredFences.ts

```diff
@@ -0,0 +1,120 @@
+import reportError from '../core/reportError';
+import * as glob from 'glob';
+import * as path from 'path';
+import getAllConfigs from '../utils/getAllConfigs';
+import Config from '../types/config/Config';
+import ConfigSet from '../types/ConfigSet';
+
+interface ConfigPathPair {
+    path: string;
+    config: Config;
+}
+
+interface RequiredFenceGlobInfo {
+    configPath: string;
+    requiredFenceGlob: string;
+}
+
+interface RequiredFence {
+    requiredPath: string;
+    configPath: string;
+}
+
+interface RequiredFenceMap {
+    [index: string]: RequiredFence[];
+}
+
+
+export default function validateRequiredFences() {
+    const configPathMap = getAllConfigs();
+    const requiredFencePairs = getRequiredFenceGlobs(configPathMap);
+    
+    // create a map from (requiredFence) => configsWithRequirement[]
+    const requiredFenceMap = getRequiredFenceMap(requiredFencePairs);
+
+    // get the required paths that aren't in our map of all configs
+    const missingRequiredPaths = Object.keys(requiredFenceMap).filter(rf => !configPathMap[rf]);
+
+    if (missingRequiredPaths.length > 0) {
+        missingRequiredPaths.map(p => ({key: p, values: requiredFenceMap[p]})).forEach(kvp => {
```

Comment:
>map(p => ({key: p, values: requiredFenceMap[p]})) [](start = 29, length = 49)

Why not just look up values inside the `forEach`?  Seems overcomplicated to iterate twice and use intermediate data structures.